src/Sulu/Component/HttpKernel/SuluKernel.php

```diff
@@ -125,7 +124,7 @@ protected function generateContainerClass($class)
         return $class;
     }
 
-    protected function configureRoutes(RouteCollectionBuilder $routes)
+    protected function configureRoutes(RoutingConfigurator $routes)
```

Comment:
The `4.4` micro kernel seems still integrate this (https://github.com/symfony/symfony/blob/30dd60857692d182aaa0a0d38ee69cb2848fdd3b/src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php#L33). And the `RoutingConfigurator` seems not to extend from it: https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Routing/RouteCollectionBuilder.php So this can lead to some issues.

I think we need to go here with a `phpdoc` like it was original done here (https://github.com/sulu/sulu/pull/5188) to avoid that the deprecated class is loaded.

Reply:
Seems like this deprecated class was added again in a false resolved merge conflict. https://github.com/sulu/sulu/commit/7e5f7ea02d7eb207edb688122a30ec7c19cc07e8#diff-6aaa8aaea84882e3b2580d844f6b1cdacbf4fea47c6950e011cb3c41c38acdbb So I would here go with the original solution with the phpdoc to avoid problems with the typehint.